src/bp-moderation/bp-moderation-filters.php

```diff
@@ -523,3 +523,67 @@ function bb_moderation_suspend_after_delete( $recode ) {
 
 }
 add_action( 'suspend_after_delete', 'bb_moderation_suspend_after_delete' );
+
+/**
+ * Function to clear the cache data on item suspend.
+ *
+ * @since BuddyBoss 1.5.6
+ *
+ * @param array $moderation_data moderation item data.
+ */
+function bb_moderation_clear_suspend_cache( $moderation_data ) {
+	if ( empty( $moderation_data['item_type'] ) || empty( $moderation_data['item_id'] ) ) {
+		return;
+	}
+	wp_cache_delete( 'bb_check_moderation_' . $moderation_data['item_type'] . '_' . $moderation_data['item_id'], 'bb' );
+	wp_cache_delete( 'bb_check_hidden_content_' . $moderation_data['item_type'] . '_' . $moderation_data['item_id'], 'bb' );
+	wp_cache_delete( 'bb_check_suspended_content_' . $moderation_data['item_type'] . '_' . $moderation_data['item_id'], 'bb' );
+	wp_cache_delete( 'bb_check_user_suspend_' . $moderation_data['item_type'] . '_' . $moderation_data['item_id'], 'bb' );
+}
+
+add_action( 'bp_suspend_before_add_suspend', 'bb_moderation_clear_suspend_cache' );
+add_action( 'bp_suspend_before_remove_suspend', 'bb_moderation_clear_suspend_cache' );
+
+/**
+ * Function to clear cache on suspend item delete.
+ *
+ * @since BuddyBoss 1.5.6
+ *
+ * @param object $suspend_record suspend item record.
+ */
+function bp_moderation_clear_delete_cache( $suspend_record ) {
```

Comment:
Update the function name: 

```suggestion
function bb_moderation_clear_delete_cache( $suspend_record ) {
```